src/ApiService/ApiService/Functions/ReproVmss.cs

```diff
@@ -43,7 +43,7 @@ public class ReproVmss {
             return response;
         }
 
-        var vms = await _context.ReproOperations.SearchStates(VmStateHelper.Available).Select(vm => vm with { Auth = null }).ToListAsync();
+        var vms = _context.ReproOperations.SearchStates(VmStateHelper.Available).Select(vm => vm with { Auth = null });
         var response2 = req.CreateResponse(HttpStatusCode.OK);
         await response2.WriteAsJsonAsync(vms);
```

Comment:
I think that this isn't going to write the correct result when it's given an `IAsyncEnumerable`

Reply:
It works with `IAsyncEnumerable` because System.Text.Json [supports it](https://docs.microsoft.com/en-us/dotnet/core/compatibility/serialization/6.0/iasyncenumerable-serialization#new-behavior)

Reply:
> It works with `IAsyncEnumerable` because System.Text.Json [supports it](https://docs.microsoft.com/en-us/dotnet/core/compatibility/serialization/6.0/iasyncenumerable-serialization#new-behavior)

nice!